collectors/ebpf.plugin/ebpf_oomkill.c

```diff
@@ -0,0 +1,203 @@
+        .user_input = 0,
+        .type = NETDATA_EBPF_MAP_STATIC,
+        .map_fd = ND_EBPF_MAP_FD_NOT_INITIALIZED
+    },
+    /* end */
+    {
+        .name = NULL,
+        .internal_input = 0,
+        .user_input = 0,
+        .type = NETDATA_EBPF_MAP_CONTROLLER,
+        .map_fd = ND_EBPF_MAP_FD_NOT_INITIALIZED
+    }
+};
+
+static ebpf_data_t oomkill_data;
+
+static ebpf_tracepoint_t oomkill_tracepoints[] = {
+    {.enabled = false, .class = "oom", .event = "mark_victim"},
+    /* end */
+    {.enabled = false, .class = NULL, .event = NULL}
+};
+
+static struct bpf_link **probe_links = NULL;
+static struct bpf_object *objects = NULL;
+
+// tmp store for oomkill values we get from a per-CPU eBPF map.
+static oomkill_ebpf_val_t *oomkill_ebpf_vals = NULL;
+
+/**
+ * Clean up the main thread.
+ *
+ * @param ptr thread data.
+ */
+static void oomkill_cleanup(void *ptr)
+{
+    ebpf_module_t *em = (ebpf_module_t *)ptr;
+    if (!em->enabled) {
+        return;
+    }
+
+    freez(oomkill_ebpf_vals);
+
+    if (probe_links) {
+        struct bpf_program *prog;
+        size_t i = 0 ;
+        bpf_object__for_each_program(prog, objects) {
+            bpf_link__destroy(probe_links[i]);
+            i++;
+        }
+        bpf_object__close(objects);
+    }
+}
+
+static void oomkill_write_data()
+{
+    uint32_t curr_key = 0;
+    uint32_t key = 0;
+    int mapfd = oomkill_maps[OOMKILL_MAP_KILLCNT].map_fd;
+    while (bpf_map_get_next_key(mapfd, &curr_key, &key) == 0) {
+        curr_key = key;
+
+        int test;
+
+        // get val for this key.
+        test = bpf_map_lookup_elem(mapfd, &key, oomkill_ebpf_vals);
+        if (unlikely(test < 0)) {
+            continue;
+        }
+
+        // now delete it, since we have a user-space copy, and never need this
+        // entry again. there's no race here, because the PID will only get OOM
+        // killed once.
+        test = bpf_map_delete_elem(mapfd, &key);
+        if (unlikely(test < 0)) {
+            // since there's only 1 thread doing these deletions, it should be
+            // impossible to get this condition.
+            error("key unexpectedly not available for deletion.");
+        }
+
+        // get command name from PID.
+        char comm[MAX_COMPARE_NAME+1];
+        test = get_pid_comm(key, sizeof(comm), comm);
+        if (test == -1) {
+            continue;
+        }
+
+        // write dim.
+        info("comm=%s", comm);
+        ebpf_write_global_dimension(
+            comm, comm,
+            ebpf_algorithms[NETDATA_EBPF_ABSOLUTE_IDX]
+        );
+        write_chart_dimension(comm, 1);
+    }
+}
+
+static void oomkill_create_charts()
+{
+    ebpf_create_chart(
+        NETDATA_EBPF_APPS_GROUP,
+        "oomkills",
+        "OOM kills",
+        EBPF_COMMON_DIMENSION_KILLS,
+        "mem",
+        NULL,
+        NETDATA_EBPF_CHART_TYPE_LINE,
+        20020,
+        NULL, NULL, 0,
+        NETDATA_EBPF_MODULE_NAME_OOMKILL
+    );
```

Comment:
When we are creating charts here, we are not adding any dimensions, and this is completely different of all other charts we have right now in `apps` menu. I think this will be strange for our users. Following the current logic that we have, I think the users will expect that all  dimensions defined at `apps_groups.conf` will have `0` when an `oom` did not happen, and they will have a value different of zero when an event happens. 

We also could have a global chart show total of `oom` that are happening on machine. :thinking:

Reply:
I think the best plan is then to just move this chart next to the existing OOM chart `mem.oom_kill`, which only shows the raw count of OOM kills on the machine.

> We also could have a global chart show total of oom that are happening on machine. thinking

This chart already exists:

![image](https://user-images.githubusercontent.com/10018065/131934612-cd55cd9a-7115-4f76-870b-91ee9f0066a8.png)

What do you think?

Reply:
This is a good approach, and I think we can move forward with this, and if it is necessary to bring something integrated with `apps`, we follow the pattern that we already have.

Reply:
Ok perfect, I'll just move the chart then and it should be good. This is actually a really useful chart, but only in those very rare & desperate times!